keras_retinanet/utils/eval.py

```diff
@@ -330,8 +392,16 @@ def evaluate(
         precision = true_positives / np.maximum(true_positives + false_positives, np.finfo(np.float64).eps)
 
         # compute average precision
-        average_precision  = _compute_ap(recall, precision)
-        mean_precisions[label] = precision.mean(), num_annotations
-        average_precisions[label] = average_precision, num_annotations
+        average_precision  = _compute_ap(recall, precision, metrics)
+
+        for metric in metrics:
+            metrics_values[metric] = {}
+            if metric == 'precision':
+                metrics_values[metric][label] = precision.mean(), num_annotations
+            else:
+                metrics_values[metric][label] = average_precision[metric], num_annotations
+
+    for metric in metrics:
```

Comment:
Do you really need these loop. I think that you already have all data you need in the **matrics_values** variable and can just return it. Next to lines doesn't make any sense.